حل مشکل سال 1404 به دلیل عدم محاسبه صحیح تاریخ 1403/12/30

Gholomreza opened the pull request:
Fix 1404 Persian Year

Gholomreza committed:
Update algorithms.js

Fix 1404 Persian Year

babakhani commented:
Hi @Gholomreza,

Thank you for your contribution! Could you please share the source of the new algorithm you’ve implemented? Additionally, I’d appreciate it if you could verify your algorithm against the data provided in this document:

[Kabise Shamsi 1206-1498](https://calendar.ut.ac.ir/documents/2139738/7092644/Kabise+Shamsi+1206-1498.pdf/fbc45bf4-df46-c298-381c-2bff2c57f0e0?t=1719740029743)

After reviewing the options, I believe we should adopt the model used in Jalaali.js, which is based on the following research:

[Persian Calendar Algorithm (EMP)](https://www.astro.uni.torun.pl/~kb/Papers/EMP/PersianC-EMP.htm)

To streamline our implementation, I’ve created a `dev` branch where I’ve removed the older algorithms (astronomical and algorithmic) and integrated the Jalaali.js model instead.

Could you review the dev branch and let me know if merging your PR is still necessary? Your feedback would be greatly appreciated.

Thanks again for your collaboration!

Gholomreza commented:
سلاممتوجه الگوريتم ها هستمهدف بنده حل سريع مورد مذكور در برنچ 0.1.8 بودهكما اينكه همين تغييرات در پروژه ديگرى تاييد شدحل مشکل سال 1404 به دلیل عدم محاسبه صحیح تاریخ 1403/12/30 by Gholomreza · Pull Request #32 · AminRahimi/angular-bootstrap-persian-datepickergithub.comSent from my iPhoneOn Mar 25, 2025, at 2:22 PM, Reza Babakhani ***@***.***> wrote:﻿
  Hi @Gholomreza,
Thank you for your contribution! Could you please share the source of the new algorithm you’ve implemented? Additionally, I’d appreciate it if you could verify your algorithm against the data provided in this document:
Kabise Shamsi 1206-1498
After reviewing the options, I believe we should adopt the model used in Jalaali.js, which is based on the following research:
Persian Calendar Algorithm (EMP)
To streamline our implementation, I’ve created a dev branch where I’ve removed the older algorithms (astronomical and algorithmic) and integrated the Jalaali.js model instead.
Could you review the dev branch and let me know if merging your PR is still necessary? Your feedback would be greatly appreciated.
Thanks again for your collaboration!—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>



babakhani left a comment (babakhani/PersianDate#152)
Hi @Gholomreza,
Thank you for your contribution! Could you please share the source of the new algorithm you’ve implemented? Additionally, I’d appreciate it if you could verify your algorithm against the data provided in this document:
Kabise Shamsi 1206-1498
After reviewing the options, I believe we should adopt the model used in Jalaali.js, which is based on the following research:
Persian Calendar Algorithm (EMP)
To streamline our implementation, I’ve created a dev branch where I’ve removed the older algorithms (astronomical and algorithmic) and integrated the Jalaali.js model instead.
Could you review the dev branch and let me know if merging your PR is still necessary? Your feedback would be greatly appreciated.
Thanks again for your collaboration!

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
